### PR TITLE
Add Academic Torrents search plugin

### PR DESCRIPTION
**Pull request by solenskiner**
Academic Torrents is a distributed system for sharing enormous datasets - for researchers, by researchers.

### CODE REVIEW

File: nova3/engines/academictorrents.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        print(download_file(info))
+
+    def search(self, what, cat='all'):
+        """ Performs search """
+        response = None
+        if cat == 'all':
+            response = retrieve_url(self.search_url.format(what=what))
+        else:
+            response = retrieve_url(self.categorysearch_url.format(
+                what=what, cat=supported_categories[cat]
```
Comment (Chocobo1) [inline nova3/engines/academictorrents.py]: CI fails at this line.

**Comment by Chocobo1** [comment]
When you're ready for review, squash commits into 1 and ping me again.
Because this repo has little manpower, you cannot expect to update it often.

I also would expect seeders & leechers (downloaders) fields to be present and results from latter pages... yeah I just tested it.

**Comment by solenskiner** [comment]
>When you're ready for review, squash commits into 1 and ping me again. [...] I also would expect seeders & leechers (downloaders) fields to be present and results from latter pages

all done. I defaulted seeds & leech to 0 for those cases when those table cells are empty on the web page.

**Comment by Chocobo1** [comment]
I found some issues:

1. The version string displayed in qbittorrent install search plugin window shows `0.0`. I guess removing the first space in `# VERSION: 0.3` in the first line can fix this.
2. After installing this plugin, there are a lot of blank entries in category selection combobox. 
   qbt only allow predefined categories to be used, this is a design decision.

**Comment by solenskiner** [comment]
1# Oh, I didn't notice. Fixed locally.

Re #2, I dont feel that the AcademicTorrent categories match to any of the predefined categories. How do you think I should fix this? Should i simply not expose AT categories, and return no hits if a QBT category is selected (since whatever keyword hit one might get, the category doesn't match)?

**Comment by Chocobo1** [comment]
>Should i simply not expose AT categories, and return no hits if a QBT category is selected (since whatever keyword hit one might get, the category doesn't match)?

Well, that is one feasible solution.

**Comment by solenskiner** [comment]
Fixed and pushed.

**Comment by Chocobo1** [comment]
Once merged, generic torrent search queries is likely to leak to "Academic Torrents" site.
I presume this won't be a problem, right?

**Comment by solenskiner** [comment]
I emailed them to ask. They said it was awesome, but felt concerns about server load, and asked if it can be disabled by default. Remind me, how does it work again, search plugins are unofficial and dont get installed by default, but rather require the user to install them, right?

**Comment by Chocobo1** [comment]
>Remind me, how does it work again, search plugins are unofficial and dont get installed by default, but rather require the user to install them, right?

If you want it to be unofficial, then you should create an entry here instead: https://github.com/qbittorrent/search-plugins/wiki/Unofficial-search-plugins

If merged into this repository, it will get installed (and enabled by default) on all qbt users who uses search functionality.

**Comment by solenskiner** [comment]
I've talked to them some more, and I think thats the best way forward. Sorry to have wasted your time.

**Pull request closed without merging**